File: java-adman-spring/application/src/main/java/com/example/adman/BusinessPartnerAddressController.java
package com.example.adman;

import com.sap.cloud.sdk.s4hana.datamodel.odata.namespaces.businesspartner.BusinessPartnerAddress;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.server.ResponseStatusException;

@RestController
public class BusinessPartnerAddressController {

    private static final Logger logger
            = LoggerFactory.getLogger(BusinessPartnerController.class);

    @PostMapping("/api/addresses")
    BusinessPartnerAddress doPost(
            @RequestBody BusinessPartnerAddress newAddress) {

        BusinessPartnerAddress bpa;
        try {
            bpa = BusinessPartnerAddressRepository.save(newAddress);
        }
        catch (
            BusinessPartnerAddressRepository_InternalErrorException e)
        {
            final String message = "Internal error " +
                    "saving the business partner address.";
            logger.error(message, e);
            throw new ResponseStatusException(
                    HttpStatus.INTERNAL_SERVER_ERROR, message, e);
        }
        return bpa;
    }

    @PatchMapping("/api/addresses")
    BusinessPartnerAddress doPatch(
            @RequestBody BusinessPartnerAddress address) {

        BusinessPartnerAddress bpa;
        try {
            bpa = BusinessPartnerAddressRepository.save(address);
        }
        catch (
            BusinessPartnerAddressRepository_InternalErrorException e)
        {
            final String message = "Internal error " +
                    "updating the business partner address.";
            logger.error(message, e);
            throw new ResponseStatusException(
                    HttpStatus.INTERNAL_SERVER_ERROR, message, e);
        }
        return bpa;
    }

    @DeleteMapping("/api/addresses")
    protected void doDelete(
            @RequestParam String businessPartnerId,
            @RequestParam String addressId) {
        try {
            BusinessPartnerAddressRepository
                    .delete(businessPartnerId, addressId);
        }
        catch (
            BusinessPartnerAddressRepository_InternalErrorException e)
        {
            final String message = "Internal error " +
                    "deleting the business partner address.";
            logger.error(message, e);
            throw new ResponseStatusException(
                    HttpStatus.INTERNAL_SERVER_ERROR, message, e);
        }
    }
}

File: ts-adman/src/adman/business-partner-address.controller.ts
import { Controller, Post, Body, Patch, Delete, Query } from '@nestjs/common';
import { BusinessPartnerAddressService } from './business-partner-address.service';
import { businessPartnerService } from '@sap/cloud-sdk-vdm-business-partner-service';

const { businessPartnerAddressApi } = businessPartnerService();

@Controller('api/addresses')
export class BusinessPartnerAddressController {

  @Post()
  async postBusinessPartnerAddress( @Body() body ) {
    let bpa = businessPartnerAddressApi.entityBuilder().fromJson(body); //BusinessPartnerAddress.builder().fromJson(body);
    await BusinessPartnerAddressService.create(bpa)
      .catch(error => {
        console.log(`Failed to create business partner address - ${error.message}`);
      });
  }

  @Patch()
  async patchBusinessPartnerAddress( @Body() body ) {
    let bpa = businessPartnerAddressApi.entityBuilder().fromJson(body); //BusinessPartnerAddress.builder().fromJson(body);
    await BusinessPartnerAddressService.update(bpa)
      .catch(error => {
        console.log(`Failed to update business partner address ` 
            + `${bpa.addressId} of business partner ${bpa.businessPartner} ` 
            + ` - ${error.message}`);
      });
    }

    @Delete()
    async deleteBusinessPartnerAddress(
      @Query('businessPartnerId') businessPartnerId: string,
      @Query('addressId') addressId: string) {
      
        await BusinessPartnerAddressService.delete(businessPartnerId, addressId)
          .catch(error => {
            console.log(`Failed to update business partner address ` 
                + `${addressId} of business partner ${businessPartnerId} ` 
                + ` - ${error.message}`);
      });

    }

}
